thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -470,7 +470,9 @@ public Response updateAnomalyFunction(@NotNull @PathParam("id") Long id,
       @QueryParam("filters") String filters,
       @QueryParam("properties") String properties,
       @QueryParam("isActive") Boolean isActive,
-      @QueryParam("frequency") String frequency) throws Exception {
+      @QueryParam("frequency") String frequency,
```

Comment:
something like this:

   // modify dataGranularity, the format of input should be like '1_MINUTES', '1_HOURS', '1_DAYS'
    if (StringUtils.isNotEmpty(userInputDataGranularity)) {
      TimeGranularity dataGranularity = TimeGranularity.fromString(userInputDataGranularity);
      anomalyFunctionSpec.setBucketSize(dataGranularity.getSize());
      anomalyFunctionSpec.setBucketUnit(dataGranularity.getUnit());
    }